screen-categories/src/main/java/com/ivy/categories/CategoriesViewModel.kt

```diff
@@ -42,21 +44,64 @@ class CategoriesViewModel @Inject constructor(
     private val trnsWithRangeAndAccFiltersAct: TrnsWithRangeAndAccFiltersAct,
     private val categoryIncomeWithAccountFiltersAct: CategoryIncomeWithAccountFiltersAct,
     private val categoryWriter: CategoryWriter,
-) : FRPViewModel<CategoriesScreenState, Nothing>() {
+) : ComposeViewModel<CategoriesScreenState, CategoriesScreenEvent>() {
+
+    private val baseCurrency = mutableStateOf("")
+    private val categories = mutableStateOf<ImmutableList<CategoryData>>(persistentListOf<CategoryData>())
+    private val reorderModalVisible = mutableStateOf(false)
+    private val categoryModalData = mutableStateOf<CategoryModalData?>(null)
+    private val sortModalVisible = mutableStateOf(false)
+    private val sortOrder = mutableStateOf(SortOrder.DEFAULT)
+
+    @Composable override fun uiState(): CategoriesScreenState {
+        LaunchedEffect(key1 = Unit) {
+            start()
+        }
+
+        return CategoriesScreenState(
+          baseCurrency = getBaseCurrency(),
+          categories = getCategories(),
+          reorderModalVisible = getReorderModalVisible(),
+          categoryModalData = getCategoryModalData(),
+          sortOrder = getSortOrder(),
+          sortModalVisible = getSortModalVisible()
+        )
+    }
 
-    override val _state: MutableStateFlow<CategoriesScreenState> = MutableStateFlow(
-        CategoriesScreenState()
-    )
+    @Composable
+    private fun getBaseCurrency(): String {
+        return baseCurrency.value
+    }
 
-    override suspend fun handleEvent(event: Nothing): suspend () -> CategoriesScreenState {
-        TODO("Not yet implemented")
+    @Composable
+    private fun getCategories(): ImmutableList<CategoryData> {
+        return categories.value
+    }
+
+    @Composable
+    private fun getReorderModalVisible(): Boolean {
+        return reorderModalVisible.value
+    }
+
+    @Composable
+    private fun getCategoryModalData(): CategoryModalData? {
+        return categoryModalData.value
+    }
+
+    @Composable
+    private fun getSortOrder(): SortOrder {
+        return sortOrder.value
+    }
+
+    @Composable
+    private fun getSortModalVisible(): Boolean {
+        return sortModalVisible.value
     }
 
     private var allAccounts = emptyList<Account>()
-    private var baseCurrency = ""
     private var transactions = emptyList<Transaction>()
 
-    fun start() {
+    private fun start() {
```

Comment:
That's out of the scope but can you also remove the `TestIdlingResource` it's a legacy thing that's no longer needed. Just delete its calls in the class.